libs/dh/market-participant/actors/feature-actors/src/lib/create/dh-actors-create-actor-modal.component.ts

```diff
@@ -238,4 +248,85 @@ export class DhActorsCreateActorModalComponent {
+              lastName: this.newUserForm.controls.lastName.value,
+              phoneNumber: this.newUserForm.controls.phone.value,
+              assignedRoles: this._userSelectedUserRoles.map((userRole) => userRole.id),
+              assignedActor: 'f3df856f-bd11-4174-97cb-fb6bc54c300a',
+            },
+            organizationId: this.chooseOrganizationForm.controls.orgId.value,
+            organization: this.chooseOrganizationForm.controls.orgId.value
+              ? null
+              : {
+                  address: {
+                    country: this.newOrganizationForm.controls.country.value,
+                  },
+                  domain: this.newOrganizationForm.controls.domain.value,
+                  businessRegisterIdentifier: this.newOrganizationForm.controls.cvrNumber.value,
+                  name: this.newOrganizationForm.controls.companyName.value,
+                },
+            actorContact: {
+              email: this.newActorForm.controls.contact.controls.email.value,
+              name: this.newActorForm.controls.contact.controls.departmentOrName.value,
+              phone: this.newActorForm.controls.contact.controls.phone.value,
+              category: ContactCategoryType.Default,
+            },
+            actor: {
+              name: { value: this.newActorForm.controls.name.value },
+              organizationId:
+                this.chooseOrganizationForm.controls.orgId.value ??
+                'c3df856f-bd11-4174-97cb-fb6bc54c300b',
+              marketRoles: [
+                {
+                  eicFunction: this.newActorForm.controls.marketrole.value,
+                  gridAreas: this.newActorForm.controls.gridArea.value
+                    ? [{ id: this.newActorForm.controls.gridArea.value, meteringPointTypes: [] }]
+                    : [],
+                },
+              ],
+              actorNumber: {
+                value: this.newActorForm.controls.glnOrEicNumber.value,
+              },
+            },
+          },
+        },
+      })
+      .subscribe((result) => this.handleCreateMarketParticipentResponse(result));
+  }
+
+  private handleCreateMarketParticipentResponse(
+    response: MutationResult<CreateMarketParticipantMutation>
+  ): void {
+    if (response.errors && response.errors.length > 0) {
+      this._toastService.open({
+        type: 'danger',
+        message: parseGraphQLErrorResponse(response.errors),
+      });
+    }
+
+    if (
+      response.data?.createMarketParticipant?.errors &&
+      response.data?.createMarketParticipant?.errors.length > 0
+    ) {
+      this._toastService.open({
+        type: 'danger',
+        message: parseApiErrorResponse(response.data?.createMarketParticipant?.errors),
+      });
+    }
+
+    if (response.data?.createMarketParticipant?.success) {
+      this._toastService.open({ type: 'success', message: 'Market participant created' });
+    }
+
+    this.close();
```

Comment:
question: if the create query/mutation errors, this will also close the modal. Is this desired? 
Can an error be recovered from and the whole form submitted again, or it must start fresh?